lucene/core/src/java/org/apache/lucene/util/automaton/IntSet.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.util.automaton;
+
+import java.util.Arrays;
+
+public abstract class IntSet {
+    int[] values;
+
+    int hashCode;
+
+    /**
+     * @return the number of values contained within this set
+     */
+    abstract int size();
+
+    @Override
+    public int hashCode() {
+        return hashCode;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == null) {
+            return false;
+        }
+        if (!(other instanceof IntSet)) {
+            return false;
+        }
+        IntSet intSet = (IntSet) other;
+        if (hashCode != intSet.hashCode) {
+            return false;
+        }
+        return Arrays.equals(values, 0, size(), intSet.values, 0, intSet.size());
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder().append('[');
```

Comment:
If it's just for debugging then I'd just use a stream and join the values there, it's a one-liner. Space after for(